torch/ao/quantization/pt2e/representation/rewrite.py

```diff
@@ -207,30 +208,81 @@ def _reference_dequantize_per_tensor_int8(x_i8, scale, zero_point, quant_min, qu
     # TODO: debug the implementation later when torchdynamo time out issue is resolved
     return ((x_i8.to(torch.float32) - zero_point) * scale).to(dtype=torch.float32)
 
-# boolean flag to indicate if we want to replace literals to be placeholder nodes for pattern and replacement
-_REPLACE_LITERAL = True
-_DONT_REPLACE_LITERAL = False
+_QUANTIZE_PER_CHANNEL_INT8_EXAMPLE_INPUTS = (
+    torch.randn(1, 3, 3, 3, dtype=torch.float),
+    torch.randn(3, dtype=torch.float),
+    torch.zeros(3, dtype=torch.int),
+    1,
+    -128,
+    127,
+)
+
+def _quantize_per_channel_int8(x_fp32, scales, zero_points, ch_axis, quant_min, quant_max):
+    out_i8 = torch.ops.quantized_decomposed.quantize_per_channel(x_fp32, scales, zero_points, ch_axis, quant_min, quant_max, torch.int8)
+    return out_i8
+
+def _reference_quantize_per_channel_int8(x_fp32, scales, zero_points, ch_axis, quant_min, quant_max):
+    x_fp32 = torch.transpose(x_fp32, ch_axis, -1)
+    out_i32 = torch.ops.aten.clamp(torch.round(x_fp32 / scales).to(torch.int32) + zero_points, quant_min, quant_max)
+    out_i32 = torch.transpose(out_i32, ch_axis, -1)
+    return out_i32.to(torch.int8)
+
+_DEQUANTIZE_PER_CHANNEL_INT8_EXAMPLE_INPUTS = (
+    torch.randint(-128, 127, (1, 3, 3, 3), dtype=torch.int8),
+    torch.randn(3, dtype=torch.float),
+    torch.zeros(3, dtype=torch.int),
+    1,
+    -128,
+    127,
+)
+
+def _dequantize_per_channel_int8(x_i8, scales, zero_points, ch_axis, quant_min, quant_max):
+    # the following will be replaced as placeholders
+    out_fp32 = torch.ops.quantized_decomposed.dequantize_per_channel(x_i8, scales, zero_points, ch_axis, quant_min, quant_max, torch.int8)
+    return out_fp32
+
+def _reference_dequantize_per_channel_int8(x_i8, scales, zero_points, ch_axis, quant_min, quant_max):
+    # the following will be replaced as placeholders
+    # in order to preserve the quant_min/quant_max args for pattern matching (e.g. matching for int4 quantized ops)
+    # we call a torch.ops.aten.clamp here
+    x_i8 = torch.ops.aten.clamp(x_i8, quant_min, quant_max)
+    x_i8 = torch.transpose(x_i8, ch_axis, -1)
+    x_i32 = x_i8.to(torch.int32)
+    out_fp32 = (x_i32 - zero_points) * scales
+    out_fp32 = torch.transpose(out_fp32, ch_axis, -1)
+    return out_fp32
+
+def _replace_ph_qdq_per_channel_replacement(gm: torch.fx.GraphModule):
+    return replace_literals_with_existing_placeholders(
+        gm,
+        exclude_literals=[-1],
+        literal_to_ph_idx={1: 3, -128: 4, 127:5}
```

Comment:
This seems really hacky. I feel it's actually clearer to just do the manual arg copying if we have to do something like this. How `replace_literals_with_placeholders` works is already a bit complicated to begin with

Reply:
I feel we do not have other solutions before we have a proper fix, and this should be easier than manual arg copying since we do not need to write additional code, how is manual arg copying easier?

Reply:
I do feel the same as Andrew, but @andrewor14 can you write an alternate solution? Then we can get an idea whether it makes this any easier. Yes it does look hacky, but also without having alternate solution, it makes it a little harder to compare.